File: vm/tests/all/interopSingleBasic/arrays1.js
//JS

llvmFile = Polyglot.evalFile("llvm", "arrays1.so");

arr = llvmFile.getSquaredArray(4);
console.log("[JS] LLVM-Array with length 4 (actual: ", arr.length, "), and arr[3] = 9 (actual: ", arr[3], ")");

File: vm/tests/all/interopSingleBasic/arrays1.cpp
//CPP
#include<polyglot.h>

void* getSquaredArray(int n) {
	int arr[n];
	for(int i=0;i<n;i++) {
		arr[i] = i*i;
	}
	void* ret = polyglot_from_i64_array(arr, n);
	return ret;
}


